src/registryctl/api/registry/blob/blob.go

```diff
@@ -33,14 +40,19 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 // DeleteBlob ...
 func (h *handler) delete(w http.ResponseWriter, r *http.Request) {
+	ctx, span := tracelib.StartTrace(r.Context(), tracerName, "delete-blob", trace.WithAttributes(attribute.Key("method").String(r.Method)))
+	defer span.End()
 	ref := mux.Vars(r)["reference"]
 	if ref == "" {
-		api.HandleBadRequest(w, errors.New("no reference specified"))
+		err := errors.New("no reference specified")
+		tracelib.RecordError(span, err, "no reference specified")
+		api.HandleBadRequest(w, err)
 		return
 	}
 	// don't parse the reference here as RemoveBlob does.
-	cleaner := storage.NewVacuum(r.Context(), h.storageDriver)
+	cleaner := storage.NewVacuum(ctx, h.storageDriver)
```

Comment:
is the ctx same as r.Context()?

Reply:
yes, it's source ctx is `r.Context()`,  and `tracelib.StartTrace(r.Context()...` added some more info in it